network/netcore/src/transport/tcp.rs

```diff
@@ -105,8 +109,29 @@ impl Transport for TcpTransport {
             .or_else(|| parse_dns_tcp(protos).map(|_| ()))
             .ok_or_else(|| invalid_addr_error(&addr))?;
 
+        let proxy_addr = match std::env::var("https_proxy") {
+            Ok(https_proxy) => match Url::parse(&https_proxy) {
+                Ok(url) => {
+                    if url.has_host() && url.scheme() == "http" {
+                        Some(format!(
+                            "{}:{}",
+                            url.host().unwrap(),
+                            url.port_or_known_default().unwrap()
+                        ))
+                    } else {
+                        None
+                    }
+                }
+                Err(_e) => None,
+            },
+            Err(_e) => None,
```

Comment:
Nit: Any reason not to just put `_`?  Looks like we're allowing for `https_proxy` but not being strict that all connections must come from it then?